modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -53,18 +45,18 @@
 (driver/register! :mongo)
 
 (defmethod driver/can-connect? :mongo
-  [_ details]
-  (with-mongo-connection [^DB conn, details]
-    (let [db-stats (-> (cmd/db-stats conn)
-                       (m.conversion/from-db-object :keywordize))
-          db-names (mg/get-db-names mongo.util/*mongo-client*)]
-      (and
-       ;; 1. check db.dbStats command completes successfully
-       (= (float (:ok db-stats))
-          1.0)
-       ;; 2. check the database is actually on the server
-       ;; (this is required because (1) is true even if the database doesn't exist)
-       (contains? db-names (:db db-stats))))))
+  [_ db-details]
+  (mongo.connection/with-mongo-client [^MongoClient c db-details]
+    (let [db-names (mongo.util/list-database-names c)]
+      (mongo.connection/with-mongo-database [^MongoDatabase db db-details]
```

Comment:
`MongoDatabase` class does not provide getter for `MongoClient` it was created with. `with-mongo-database` is meant as utility to avoid `(let [db (mongo.util/database client ...)` in calls of `with-mongo-client`, because we are using database without the need for client in various places.

As we need both in this function, I've modified the code to use the `let` form.